webgl/penworld.js

```diff
@@ -624,3 +624,87 @@ function render () {
 
   window.requestAnimationFrame(render)
 }
+
+Vector3.prototype.listify = function() { // compact
+  return [this.x, this.y, this.z];
+};
+
+Vector3.prototype.fromList = function(l) { // extract
+  this.x = l[0];
+  this.y = l[1];
+  this.z = l[2];
+}
+
+function save() {
+  
+  const data = Subchunk.subchunks; // world data
+  const chunks = Object.keys(data); // chunks
+  let result = {}; // resulting data
+  
+  for (let i=0; i<chunks.length; i++) { // iterate over chunks
+    const index = chunks[i];
+    const chunk = data[index]; // choose current chunk
+    
+    result[index] = [chunk.pos.listify(), []]; // converts Vector3 to list
+    
+    const blocks = chunk.blocks; // get blocks
+    for (let j=0; j<blocks.length; j++) { // iterate over blocks
+      const block = blocks[j]; // choose current block
+      
+      // don't store chunk or faces because they're redundant
+      if (block !== null) { // if it's not null
+        result[index][1].push([block.pos.listify(), block.type]);
+      } else { // bad idea but i'm lazy and i'll work on it tomorrow
+        result[index][1].push(0);
+      }
+    }
+  }
+  
+  result.version = 1;
+  
+  return JSON.stringify(result);
+}
+
+function load(data) {
+  data = JSON.parse(data);
+  delete data.version;
+  const chunks = Object.keys(data); // chunks
+  s = data;
+  Subchunk.subchunks = {};
```

Comment:
I don't think this line is necessary if you set `Subchunk.subchunks` again later in the function